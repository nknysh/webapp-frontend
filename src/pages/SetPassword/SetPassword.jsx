import React, { Fragment, useCallback } from 'react';
import { Modal } from '@pure-escapes/webapp-ui-components';

import { SetPasswordForm } from 'containers';
import Home from 'pages/Home';

import { propTypes } from './SetPassword.props';

export const SetPassword = ({ history, match, ...props }) => {
  const { params: token } = match;
  const onClose = useCallback(() => {
    history.push('/');
  }, [history]);

  return (
    <Fragment>
      <Home history={history} match={match} {...props} />
      <Modal open={true} onClose={onClose}>
        <SetPasswordForm token={token} />
      </Modal>
    </Fragment>
  );
};

SetPassword.propTypes = propTypes;

export default SetPassword;
