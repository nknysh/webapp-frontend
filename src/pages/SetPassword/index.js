import { lazy } from 'react';

export { default } from './SetPassword';

export const AsyncSetPassword = lazy(() => import('./SetPassword'));
