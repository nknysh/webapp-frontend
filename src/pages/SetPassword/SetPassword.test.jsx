import React from 'react';

import { SetPassword } from './SetPassword';

const getComponent = props => shallow(<SetPassword match={{ params: {} }} {...props} />);

describe('<SetPassword />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
