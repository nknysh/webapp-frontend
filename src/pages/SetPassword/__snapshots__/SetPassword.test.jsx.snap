// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SetPassword /> render matches snapshot 1`] = `
<Fragment>
  <React.Lazy
    match={
      Object {
        "params": Object {},
      }
    }
  />
  <Modal
    modalContentProps={Object {}}
    onClose={[Function]}
    open={true}
  >
    <Connect(SetPasswordForm)
      token={Object {}}
    />
  </Modal>
</Fragment>
`;
